RecoLocalTracker/SiPixelRecHits/src/PixelCPEFast.cc

```diff
@@ -160,21 +160,28 @@ void PixelCPEFast::fillParamsForGpu() {
 
     // errors .....
     ClusterParamGeneric cp;
-    auto gvx = p.theOrigin.x() + 40.f * commonParamsGPU_.thePitchX;
-    auto gvy = p.theOrigin.y();
-    auto gvz = 1.f / p.theOrigin.z();
-    //--- Note that the normalization is not required as only the ratio used
-
-    // calculate angles
-    cp.cotalpha = gvx * gvz;
-    cp.cotbeta = gvy * gvz;
 
     cp.with_track_angle = false;
 
     auto lape = p.theDet->localAlignmentError();
     if (lape.invalid())
       lape = LocalError();  // zero....
 
+    g.apeX = lape.xx();
+    g.apeY = lape.yy();
+
+    auto toMicron = [&](float x) { return std::min(511, int(x * 1.e4f + 0.5f)); };
+
+    {
+      // average angle
+      auto gvx = p.theOrigin.x() + 40.f * commonParamsGPU_.thePitchX;
+      auto gvy = p.theOrigin.y();
+      auto gvz = 1.f / p.theOrigin.z();
```

Comment:
why are these in a separate scope?
`p` is const at this point and the same values are needed again later (in a loop!).
Please reuse.

Reply:
All this happens just once per job. And in future this blocks may evolve.

Reply:
This is repeated a few lines later, could these be merged?
```cpp
// average angle
    auto gvx = p.theOrigin.x() + 40.f * commonParamsGPU_.thePitchX;
    auto gvy = p.theOrigin.y();
    auto gvz = 1.f / p.theOrigin.z();
    //--- Note that the normalization is not required as only the ratio used

    // calculate angles
    cp.cotalpha = gvx * gvz;
    cp.cotbeta = gvy * gvz;
```

Reply:
do. please